### PR TITLE
[KOGITO-1539] Renaming and removing duplicates

### PR DESCRIPTION
**Pull request by ricardozanini**
Guys,

I did some renaming and removed some duplications (timer examples and rule units) in order to have a more comprehensive examples directory.

jBPM -> Business Process
Drools -> Rules

See: https://issues.redhat.com/browse/KOGITO-1539

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -158,8 +155,8 @@
     <module>kogito-usertasks-with-security-oidc-springboot</module>
     <module>kogito-usertasks-with-security-quarkus</module>
     <module>kogito-usertasks-with-security-springboot</module>
-    <module>kogito-timer-quarkus</module>
-    <module>kogito-timer-springboot</module>
+    <module>timer-quarkus</module>
```
Comment (cristianonicolai) [inline pom.xml]:
```suggestion
    <module>business-process-timer-quarkus</module>
```

File: pom.xml [hunk 2]
```diff
@@ -158,8 +155,8 @@
...
+    <module>timer-springboot</module>
```
Comment (cristianonicolai) [inline pom.xml]:
```suggestion
    <module>business-process-timer-springboot</module>
```

**Comment by cristianonicolai** [comment]
@sterobin fyi this will probably cause some links to be broken in the docs

**Review by sutaakar — APPROVED** [review]

**Comment by ricardozanini** [comment]
@radtriste the operator examples should also change for 0.9.1: https://issues.redhat.com/browse/KOGITO-1669

**Review by radtriste — CHANGES_REQUESTED** [review]
Missing the business-process-timer-quarkus ?

Also, some links into the README are not updated

**Comment by evacchi** [comment]
I apologize for being the naysayer here, but would it be ok to just use "process" instead of "business-process"? Mostly because it is shorter, but also because it removes the "business" part. I would even go further and say "workflow" but I think we are already using "process" in a lot of places. wdyt

**Comment by cristianonicolai** [comment]
> I apologize for being the naysayer here, but would it be ok to just use "process" instead of "business-process"? Mostly because it is shorter, but also because it removes the "business" part. I would even go further and say "workflow" but I think we are already using "process" in a lot of places. wdyt

I'm ok with that ( renaming to process ).

**Review by ge0ffrey — APPROVED** [review]
Approved, but we have an open PR for the quarkus version of the jbpm-optaplanner example, so that will need to be renamed too. Talk to Christopher about it. I think that PR is ready to be merged.

**Comment by ricardozanini** [comment]
> Approved, but we have an open PR for the quarkus version of the jbpm-optaplanner example, so that will need to be renamed too. Talk to Christopher about it. I think that PR is ready to be merged.

I'd say that you guys can merge ahead and I'll rebase this one.

**Comment by ricardozanini** [comment]
> I apologize for being the naysayer here, but would it be ok to just use "process" instead of "business-process"? Mostly because it is shorter, but also because it removes the "business" part. I would even go further and say "workflow" but I think we are already using "process" in a lot of places. wdyt

That was discussed in the chat and @tsurdilo pointed that it's not a process, but business process the right terminology.

I also agree that would be better to have a short name in this case. You guys decide, I'm not a specialist for this terminology.

**Comment by ricardozanini** [comment]
> Missing the business-process-timer-quarkus ?
 
dammit, I haven't added in the last commit. Thanks.

> Also, some links into the README are not updated

Could you point them?

**Comment by evacchi** [comment]
well I am not an expert, I would have preferred something shorter (like workflow) but if process is incorrect let's keep it that way /cc @tsurdilo

File: pom.xml [hunk 3]
```diff
@@ -120,24 +120,21 @@
...
-    <module>drools-quarkus-unit-example</module>
-    <module>drools-quarkus-helloworld</module>
-    <module>drools-polyglot-example</module>
-    <module>jbpm-quarkus-helloworld</module>
-    <module>jbpm-quarkus-example</module>
-    <module>jbpm-springboot-example</module>
-    <module>jbpm-optaplanner-springboot-example</module>
+    <module>rules-quarkus-example</module>
+    <module>rules-quarkus-helloworld</module>
+    <module>rules-polyglot-example</module>
```
Comment (evacchi) [inline pom.xml]: i think it makes little sense to keep polyglot

File: pom.xml [hunk 4]
```diff
@@ -120,24 +120,21 @@
...
+    <module>business-process-quarkus-helloworld</module>
+    <module>business-process-quarkus-example</module>
+    <module>business-process-springboot-example</module>
+    <module>business-process-optaplanner-springboot</module>
     <module>onboarding-example</module>
-    <module>jbpm-polyglot-example</module>
+    <module>business-process-polyglot-example</module>
```
Comment (evacchi) [inline pom.xml]: same here for polyglot

File: pom.xml [hunk 5]
```diff
@@ -120,24 +120,21 @@
...
     <module>dmn-quarkus-example</module>
     <module>dmn-springboot-example</module>
     <module>ruleunit-quarkus-example</module>
```
Comment (evacchi) [inline pom.xml:154]: i think 

- rules-quarkus-example
- rules-quarkus-helloworld
- ruleunit-quarkus-example

are all very similar.
i think the last one was the most complete, authored by @tkobayas 
we may keep one other (helloworld or or rules-quarkus) and remove the others /cc @mariofusco

File: pom.xml [hunk 6]
```diff
@@ -158,8 +155,8 @@
     <module>kogito-usertasks-with-security-oidc-springboot</module>
     <module>kogito-usertasks-with-security-quarkus</module>
     <module>kogito-usertasks-with-security-springboot</module>
-    <module>kogito-timer-quarkus</module>
-    <module>kogito-timer-springboot</module>
+    <module>business-process-timer-quarkus</module>
+    <module>business-process-timer-springboot</module>
```
Comment (evacchi) [inline pom.xml]: some of these start with kogito, and some do not. I don't if it's better to prepend everything or remove from all... 

also, all these 'kogito' are usually business processes :D so either we rename all of them wiht business-process or we just keep kogito ? wdyt

**Review by evacchi — CHANGES_REQUESTED** [review]

**Comment by tsurdilo** [comment]
> I apologize for being the naysayer here, but would it be ok to just use "process" instead of "business-process"? Mostly because it is shorter, but also because it removes the "business" part. I would even go further and say "workflow" but I think we are already using "process" in a lot of places. wdyt

@evacchi @ricardozanini @cristianonicolai 
Overall I also think it's not a big deal. From my understanding, a "process" can be a visual representation only, where as a "business process" includes the "how" and "what" as well.
Also the industry has always used the *B* as in *B*PM.
 In the end if you guys think that we should shorten it let's go for it.

Just to add one more thing if I may. On all the Kogito slides we describe it as "Cloud-Native Business Automation for
building intelligent applications, backed
by battle-tested capabilities"
and "Next generation business automation toolkit"

Here we use the word "business" and not even mention "process". Just to think about it.

File: pom.xml (hunk 3 above)
Comment (ricardozanini) [reply pom.xml]: What do you suggest?

**Comment by ricardozanini** [comment]
Guys to move forward I understand that we need to come up with:

1. Replacement of `process` or `business-process` to only `kogito`.
2. Remove some `rules` and `rulesunit` examples since we have duplications in that area. ([comment](https://github.com/kiegroup/kogito-examples/pull/168#discussion_r402388868))
3. Remove `polyglot` wording or the example itself

**My opinion:**
1. Use `kogito` instead
2. @mariofusco @evacchi to decide
3. I'd keep them, but no clue about the wording

@evacchi @cristianonicolai  @mariofusco @tsurdilo could you guys chime in and vote? So I can go forward.

File: pom.xml (hunk 4 above)
Comment (ricardozanini) [reply pom.xml]: Let's remove the example itself or the wording?

File: pom.xml (hunk 5 above)
Comment (ricardozanini) [reply pom.xml:154]: Let's keep only `ruleunit-quarkus-example`?

File: pom.xml (hunk 6 above)
Comment (ricardozanini) [reply pom.xml]: Ok, so `s/business-process/kogito/g`?

**Comment by lazarotti** [comment]
kogito has been used as a synonym to business-process in most of quickstart, so yes, +1 to replace business-process to just kogito

File: business-process-quarkus-helloworld/README.md [hunk 7]
```diff
@@ -1,4 +1,4 @@
-# jBPM + Quarkus example
+# Business Process + Quarkus example
```
Comment (lazarotti) [inline business-process-quarkus-helloworld/README.md]: @ricardozanini this "hello world example" is almost the same the "kogito-scripts-*" example. However, I see this hello world does not even open in VSCode editor, it is broken. I recommend deleting it as we have the kogito-scripts-* which works, and also avoid the duplication.

**Comment by cristianonicolai** [comment]
> Guys to move forward I understand that we need to come up with:
> 
>     1. Replacement of `process` or `business-process` to only `kogito`.
> 
>     2. Remove some `rules` and `rulesunit` examples since we have duplications in that area. ([comment](https://github.com/kiegroup/kogito-examples/pull/168#discussion_r402388868))
> 
>     3. Remove `polyglot` wording or the example itself
> 
> 
> **My opinion:**
> 
>     1. Use `kogito` instead
> 
>     2. @mariofusco @evacchi to decide
> 
>     3. I'd keep them, but no clue about the wording
> 
> 
> @evacchi @cristianonicolai @mariofusco @tsurdilo could you guys chime in and vote? So I can go forward.
> 
> Depends on: #175

I think 1 and 2 have to be decided together. Kogito doesn't mean only processes, so to me is more a question of if we should keep ( consistently ) _kogito-_ as the prefix. Process/Rules/Decision/etc are the capabilities that such example showcase. It is not necessary that you append process/decision to the project name, for instance, _kogito-travel-agency_ will show many capabilities combined together.

3 - Polyglot is the term used by GraalVM, so I think we could keep it as is for now.

File: pom.xml (hunk 3 above)
Comment (evacchi) [reply pom.xml]: remove

File: pom.xml (hunk 4 above)
Comment (evacchi) [reply pom.xml]: remove

**Comment by evacchi** [comment]
summary:
- let's remove polyglot examples, they are outdated
- let's rename jbpm/process to kogito as this is the least work for now
- drools-example and hello world they are the same: let's keep helloworld
- jbpm hello world and kogito script task are the same; let's keep script task as the hello world does not work in vscode

**Comment by ricardozanini** [comment]
Updated @evacchi @cristianonicolai @lazarotti could you guys please take a look?

**Comment by krisv** [comment]
@evacchi @cristianonicolai @lazarotti @tsurdilo @mariofusco 
Please don't use kogito as the equivalent of process, imho that's kinda the opposite of what we are trying to achieve.  Kogito is the umbrella, that combines different "business models" like process, rule, constraints.
Regarding branding: we can continue to use the existing brands, like we are saying kogito is a continuation of drools/jbpm/optaplanner.  No need to go on a crusade to remove any references. 
However, (1) we do want to make it clear that kogito is completely redesigned to be cloud-native, and for that reason we are positioning a new name, Kogito, that describes this effort (to make it easy to refer to it).  Secondly, (2) we need to make it clear this is an integrated solution (not just a packaging of drools/jbpm/optaplanner) and that applications are a combination of these things together, and that they are integrated closely and share a lot.
For those two reasons, for examples / quickstarts / docs we decided we would try to describe based on capabilities (for example process + timer or process + decision) or use case (travel agency) rather than referring to them as a jbpm or drools or optaplanner example.

**Comment by evacchi** [comment]
ok so let's keep process? sorry @ricardozanini don't hate on us :(

File: kogito-timer-quarkus/README.md [hunk 8]
```diff
@@ -81,6 +81,16 @@ This needs to be given when starting process instance as delay attribute of type
 
 ## Build and run
 
+By default the [Jobs Service integration](#use-kogito-job-service-as-external-timer-service) is enabled for this example. To disabled it, just comment this dependency in the `pom.xml` file:
```
Comment (MarianMacik) [inline kogito-timer-quarkus/README.md:84]: To disable it...

File: kogito-timer-springboot/README.md [hunk 9]
```diff
@@ -75,12 +75,21 @@ This example shows
 * Cycle timer AfterTimer
 <p align="center"><img src="docs/images/timersCycleAfterTimerScriptTask.png"></p>
 
-
 Timer expression is expected to be given in ISO-8601 format e.g. PT30S - wait 30 seconds before expiring. 
 This needs to be given when starting process instance as delay attribute of type string. 
 
 ## Build and run
 
+By default the [Jobs Service integration](#use-kogito-job-service-as-external-timer-service) is enabled for this example. To disabled it, just comment this dependency in the `pom.xml` file:
```
Comment (MarianMacik) [inline kogito-timer-springboot/README.md:83]: Same here.

File: kogito-timer-springboot/pom.xml [hunk 10]
```diff
@@ -1,23 +1,19 @@
...
+<?xml version="1.0" encoding="UTF-8" ?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
     <groupId>org.kie.kogito</groupId>
     <artifactId>kogito-examples</artifactId>
     <version>8.0.0-SNAPSHOT</version>
   </parent>
 
+  <name>Kogito Examples :: Kogito Timer Springboot</name>
```
Comment (MarianMacik) [inline kogito-timer-springboot/pom.xml]: Springboot -> Spring Boot

File: kogito-optaplanner-springboot/pom.xml [hunk 11]
```diff
@@ -7,9 +7,9 @@
     <version>8.0.0-SNAPSHOT</version>
   </parent>
   <groupId>org.kie.kogito.examples</groupId>
-  <artifactId>jbpm-optaplanner-springboot-example</artifactId>
+  <artifactId>kogito-optaplanner-springboot</artifactId>
 
-  <name>Kogito Example :: jBPM, OptaPlanner and SpringBoot</name>
+  <name>Kogito Example :: Kogito, OptaPlanner and SpringBoot</name>
```
Comment (MarianMacik) [inline kogito-optaplanner-springboot/pom.xml]: Spring Boot

File: kogito-springboot-example/pom.xml [hunk 12]
```diff
@@ -7,9 +7,9 @@
     <version>8.0.0-SNAPSHOT</version>
   </parent>
   <groupId>org.kie.kogito.examples</groupId>
-  <artifactId>jbpm-springboot-example</artifactId>
+  <artifactId>kogito-springboot-example</artifactId>
 
-  <name>Kogito Example :: jBPM and SpringBoot</name>
+  <name>Kogito Example :: Kogito and SpringBoot</name>
```
Comment (MarianMacik) [inline kogito-springboot-example/pom.xml]: Spring Boot

**Review by MarianMacik — APPROVED** [review]
Just a few typos.

**Comment by ricardozanini** [comment]
I'll redo it.

**Comment by ricardozanini** [comment]
I'm closing this one since there's too much commits in the history and this branch will be deleted. Please follow up here: https://github.com/kiegroup/kogito-examples/pull/187

**Pull request closed without merging**